test/base/PromiseTest.ooc

```diff
@@ -0,0 +1,28 @@
+use ooc-base
+use ooc-unit
+use ooc-math
+import math
+import os/Time
+
+PromiseTest: class extends Fixture {
+	init: func {
+		super("Promise")
+		this add("basic", func {
+			p := ThreadedPromise< Cell<Int> > new(func -> Cell<Int> {
```

Comment:
Try not to use overly short names. `promise1` shouldn't be too many letters to write.